core/src/main/java/org/infinispan/commands/functional/Mutation.java

```diff
@@ -1,11 +1,13 @@
 package org.infinispan.commands.functional;
 
+import org.infinispan.commands.functional.functions.InjectableComponent;
+import org.infinispan.encoding.DataConversion;
 import org.infinispan.functional.EntryView;
 
 /**
  * Simplified version of functional command used for read-only operations after transactional modifications.
  */
-public interface Mutation<K, V, R> {
+public interface Mutation<K, V, R> extends InjectableComponent {
```

Comment:
Hrmm I am not a fan of having this at this level. `InjectableComponent` to me is more of an implementation specific thing.

Also should help perf a tiny little bit, since some implementations don't need injection to be called.

Reply:
@wburns I agree, but OTOH the `org.infinispan.commands.functional` package isn't `@public` either.